pkg/lightning/backend/kv/context.go

```diff
@@ -0,0 +1,260 @@
+		contextstatic.WithLocation(timeutil.SystemLocation()),
+		contextstatic.WithErrLevelMap(errLevels),
+		contextstatic.WithUserVarsReader(userVars),
+	)
+
+	// no need to build as plan cache.
+	planCacheTracker := contextutil.NewPlanCacheTracker(contextutil.IgnoreWarn)
+	intest.Assert(!planCacheTracker.UseCache())
+	ctx := contextstatic.NewStaticExprContext(
+		contextstatic.WithEvalCtx(evalCtx),
+		contextstatic.WithPlanCacheTracker(&planCacheTracker),
+	)
+
+	if len(sysVars) > 0 {
+		var err error
+		ctx, err = ctx.LoadSystemVars(sysVars)
+		if err != nil {
+			return nil, err
+		}
+		evalCtx = ctx.GetStaticEvalCtx()
+	}
+
+	currentTime := func() (time.Time, error) { return time.Now(), nil }
+	if timestamp > 0 {
+		currentTime = func() (time.Time, error) { return time.Unix(timestamp, 0), nil }
+	}
+
+	evalCtx = evalCtx.Apply(contextstatic.WithCurrentTime(currentTime))
+	ctx = ctx.Apply(contextstatic.WithEvalCtx(evalCtx))
+
+	return &litExprContext{
+		StaticExprContext: ctx,
+		userVars:          userVars,
+	}, nil
+}
+
+// SetUserVarVal sets the value of a user variable.
+func (ctx *litExprContext) SetUserVarVal(name string, dt types.Datum) {
+	ctx.userVars.SetUserVarVal(name, dt)
+}
+
+// UnsetUserVar unsets a user variable.
+func (ctx *litExprContext) UnsetUserVar(varName string) {
+	ctx.userVars.UnsetUserVar(varName)
+}
+
+// litTableMutateContext implements the `table.MutateContext` interface for lightning import.
+type litTableMutateContext struct {
+	exprCtx               *litExprContext
+	encodingConfig        tbctx.RowEncodingConfig
+	mutateBuffers         *tbctx.MutateBuffers
+	shardID               *variable.RowIDShardGenerator
+	reservedRowIDAlloc    stmtctx.ReservedRowIDAlloc
+	enableMutationChecker bool
+	assertionLevel        variable.AssertionLevel
+	tableDelta            struct {
+		sync.Mutex
+		m map[int64]variable.TableDelta
+	}
+}
+
+// AlternativeAllocators implements the `table.MutateContext` interface.
+func (*litTableMutateContext) AlternativeAllocators(*model.TableInfo) (autoid.Allocators, bool) {
+	// lightning does not support temporary tables, so we don't need to provide alternative allocators.
+	return autoid.Allocators{}, false
+}
+
+// GetExprCtx implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) GetExprCtx() exprctx.ExprContext {
+	return ctx.exprCtx
+}
+
+// ConnectionID implements the `table.MutateContext` interface.
+func (*litTableMutateContext) ConnectionID() uint64 {
+	// Just return 0 because lightning import does not in any connection.
+	return 0
+}
+
+// InRestrictedSQL implements the `table.MutateContext` interface.
+func (*litTableMutateContext) InRestrictedSQL() bool {
+	// Just return false because lightning import does not in any SQL.
+	return false
+}
+
+// TxnAssertionLevel implements the `table.MutateContext` interface.
+func (ctx *litTableMutateContext) TxnAssertionLevel() variable.AssertionLevel {
+	return ctx.assertionLevel
```

Comment:
just curious, can user change `assertionLevel` and below settings?

Reply:
I think lightning does not forbid configurationing these variables: https://github.com/pingcap/tidb/blob/4371a1d4850f8924ba922f179daa7680ef2976da/lightning/pkg/importer/get_pre_info.go#L229-L234

This PR just keep the same behavior with the previous implementation.